### DISCUSSION TITLE
Mermaid Plugin

### DISCUSSION THREAD

**Discussion by directormac** [BearToCode/carta#49] [Ideas]
Hi i'd like to request a plugin for mermaidjs!

(MermaidJS)[https://mermaid.js.org]

also congrats on 4.0 release, will use it on my next project!!

**Comment by BearToCode** [comment]
Hi! It seems like there is a Rehype [rehype](https://github.com/remcohaszing/rehype-mermaid) plugin doing exactly that. Using that should be quite easy:

```ts
import { type Plugin } from 'carta-md';
import rehypeMermaid from 'rehype-mermaid';

const mermaid: Plugin = {
  transformers: [{
    execution: 'sync',
    type: 'rehype',
    transform({ processor }) {
      processor.use(rehypeMermaid)
    }
  }]
}
```

I've not tested it, but I think it will work fine.
This repo already has many plugins and I don't thinks it's the case to add another one; if you want you can create and publish this plugin in a repo of yours and I'll link to it in the community plugins section. However, if the plugin actually consists of only these few lines of code, there might be no need to publish a package just for that.
